### PR TITLE
Added Teams addon for messaging on microsoft teams

### PR DESCRIPTION
**Pull request by ashrash290**
Feature request : #715

Added new addon
Integrating with microsoft teams

@ivarconr  Please review this PR

### CODE REVIEW

**Comment by ivarconr** [comment]
would you mind removing the "yarn.lock" change. I cant see any reason this needs to change as long as package.json remains the same.

File: docs/addons/teams.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+id: teams
+title: Microsoft Teams
+---
+
+> This feature was introduced in \_Unleash v3.11.x.
```
Comment (ivarconr) [inline docs/addons/teams.md:6]: We will bring this as part of v4.0.x

**Review by ivarconr — APPROVED** [review]

**Comment by ivarconr** [comment]
I worked on first attemt, nice!
image

**Comment by ivarconr** [comment]
A minor detail @ashrash290 , maybe we could have all activation strategies as bullets (also the first one)

image

File: docs/addons/teams.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#### Events
+
+You can choose to trigger updates for the following events (we might add more event types in the future):
+
+- feature-created
+- feature-updated
+- feature-archived
+- feature-revived
+- feature-stale-on
+- feature-stale-off
```
Comment (checketts) [inline docs/addons/teams.md:23]: I wish there was an event for a `feature-requested-that-doesnt-exist` ;) Maybe I'll add that...

**Review by checketts — APPROVED** [review]

File: docs/addons/teams.md (hunk 2 above)
Comment (ivarconr) [reply docs/addons/teams.md:23]: I think you can solve that by creating a new feature toggle with a proper description? Then the software team should just use that toggle and start creating the new feature :dancers:

Comment (checketts) [reply docs/addons/teams.md:23]: True. The only hiccup we have is that we are running 3 instances of Unleash (dev, non-prod, prod) So it is possible that an upper lane would start asking for a toggle that hasn't been created (we are still finding our preferred process)

I think you are correct, though.

**Comment by ashrash290** [comment]
> A minor detail @ashrash290 , maybe we could have all activation strategies as bullets (also the first one)
> image


Does this look good ?
Screenshot 2021-04-28 at 9 20 49 AM

File: docs/addons/teams.md (hunk 1 above)
Comment (ashrash290) [reply docs/addons/teams.md]: Done

File: src/lib/addons/teams.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        async fetchRetry(url, options, retries, backoff) {
+            this.fetchRetryCalls.push({ url, options, retries, backoff });
+            return Promise.resolve({ status: 200 });
+        }
+    },
+});
+
+const noLogger = require('../../test/fixtures/no-logger');
+
+test('Should call slack webhook', async t => {
```
Comment (chriswk) [inline src/lib/addons/teams.test.js:21]:
```suggestion
test('Should call teams webhook', async t => {
```

File: src/lib/addons/teams.test.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        url: 'http://hooks.office.com',
+    };
+
+    await addon.handleEvent(event, parameters);
+    t.is(addon.fetchRetryCalls.length, 1);
+    t.is(addon.fetchRetryCalls[0].url, parameters.url);
+    t.snapshot(addon.fetchRetryCalls[0].options.body);
+});
+
+test('Should call slack webhook for archived toggle', async t => {
```
Comment (chriswk) [inline src/lib/addons/teams.test.js:46]:
```suggestion
test('Should call teams webhook for archived toggle', async t => {
```

**Review by chriswk — APPROVED** [review]
This looks really good. Good job :)

**Pull request merged**